### PR TITLE
Fixes #3506: Disallows 0 to be entered as a price of paid ticket

### LINKED ISSUE #3506: Paid ticket allows to enter 0.00 USD or other currency
Currently it is possible to enter a paid ticket that costs 0.00 USD.

Expected: It should not be possible to enter amount 0.

### PR DESCRIPTION
**Pull request by CosmicCoder96**
fixes #3506 
image
test at: https://orega-server.herokuapp.com/

### CODE REVIEW

File: app/static/js/events/wizard/vue-app.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ var app = new Vue({
     data: {
         step: step,
         event: EVENT,
+        ticket: TICKET,
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: Why are you doing this ?

File: app/static/js/events/wizard/vue-app.js [hunk 2]
```diff
@@ -166,13 +170,13 @@ var app = new Vue({
             }
         },
         addTicket: function (ticketType) {
-            var ticket = _.cloneDeep(TICKET);
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: What's the reason behind adding ticket as a parameter in the vue-app instance itself ? javascript is pass by reference. Every time the ticket gets modified, `this.ticket` will also get modified. That's the reason there was a clone in the first place.

File: app/static/js/events/wizard/vue-app.js [hunk 3]
```diff
@@ -166,13 +170,13 @@ var app = new Vue({
...
-            ticket.sales_start_time = moment().tz(this.event.timezone).format('HH:mm');
-            ticket.sales_end_date = moment().tz(this.event.timezone).add(10, 'days').format('MM/DD/YYYY'),
-            ticket.sales_end_time = moment().tz(this.event.timezone).add(10, 'days').hour(22).minute(0).format('HH:mm'),
-            ticket.type = ticketType;
-            this.event.tickets.push(ticket);
+            this.ticket.sales_start_date = moment().tz(this.event.timezone).format('MM/DD/YYYY');
+            this.ticket.sales_start_time = moment().tz(this.event.timezone).format('HH:mm');
+            this.ticket.sales_end_date = moment().tz(this.event.timezone).add(10, 'days').format('MM/DD/YYYY'),
+            this.ticket.sales_end_time = moment().tz(this.event.timezone).add(10, 'days').hour(22).minute(0).format('HH:mm'),
+            this.ticket.price=0.01,
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: And add space around `=` to match style

File: app/static/js/events/wizard/vue-app.js [hunk 4]
```diff
@@ -166,13 +170,13 @@ var app = new Vue({
...
+            this.ticket.type = ticketType;
+            this.event.tickets.push(this.ticket);
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: You're technically pushing the same object every-time user adds a ticket

File: app/static/js/events/wizard/vue-app.js [hunk 5]
```diff
@@ -350,7 +354,9 @@ function shouldDisableMove($this) {
         $this.event.start_time_time.trim() === '' ||
         $this.event.start_time_date.trim() === '' ||
         $this.event.end_time_time.trim() ==='' ||
-        $this.event.end_time_date.trim() === ''
+        $this.event.end_time_date.trim() === '' ||
+        $this.ticket.price === '' ||
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: This is now how you're supposed to check. Each event will have multiple tickets. You'll have to loop through `this.event.tickets`

File: app/static/js/events/wizard/basic-details/initializer.js [hunk 6]
```diff
@@ -32,6 +32,7 @@ function basicDetailsInit($eventDiv) {
     $eventDiv.find(".licence-help").click(function () {
         $("#licence-list").slideToggle();
     });
+    $eventDiv.find("#ticket").val(0.01);
```
Comment (niranjan94) [inline app/static/js/events/wizard/basic-details/initializer.js]: Why are you doing this ? Please don't manually set values like this. Set the value in the appropriate object and let Vue handle the rest for you

Comment (CosmicCoder96) [inline app/static/js/events/wizard/basic-details/initializer.js]: @niranjan94  that was left unintentionally :) already have set the value in the object.

**Comment by CosmicCoder96** [comment]
@niranjan94 Thanks for the detailed review. I have made the required changes :)

**Comment by CosmicCoder96** [comment]
@niranjan94 Thanks for the detailed review. I have made the required changes :)

File: app/static/js/events/wizard/vue-app.js [hunk 7]
```diff
@@ -350,10 +358,21 @@ function shouldDisableMove($this) {
...
         $this.event.start_time_date.trim() === '' ||
         $this.event.end_time_time.trim() ==='' ||
-        $this.event.end_time_date.trim() === ''
+        $this.event.end_time_date.trim() === '' ||
+        checkTickets($this)
     );
 }
 
+function checkTickets($this) {
+    for( var ticket of $this.event.tickets){
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: Don't use ES6 syntax here. Not all browsers support it. Use lodash's foreach to loop through.

File: app/static/js/events/wizard/vue-app.js [hunk 8]
```diff
@@ -350,10 +358,21 @@ function shouldDisableMove($this) {
         $this.event.start_time_time.trim() === '' ||
         $this.event.start_time_date.trim() === '' ||
         $this.event.end_time_time.trim() ==='' ||
-        $this.event.end_time_date.trim() === ''
+        $this.event.end_time_date.trim() === '' ||
+        checkTickets($this)
     );
 }
 
+function checkTickets($this) {
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: Don't pass the entire `$this` object. Pass only the tickets.

File: app/static/js/events/wizard/vue-app.js [hunk 9]
```diff
@@ -350,10 +358,21 @@ function shouldDisableMove($this) {
...
+    for( var ticket of $this.event.tickets){
+            if(ticket.type == 'paid' && (ticket.price === '' || ticket.price <=0))
+            {
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: The logic is flawed. Even if one ticket is valid, the function exits and everything is considered valid.

File: app/static/js/events/wizard/vue-app.js [hunk 10]
```diff
@@ -95,6 +95,13 @@ var app = new Vue({
         'event.end_time_date': function () {
             this.disableMove = shouldDisableMove(this);
         },
+        'event.tickets': {
+            handler :  function () {
+                this.disableMove = shouldDisableMove(this);
+        },
+        deep: true
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: Indent this properly.

File: app/static/js/events/wizard/vue-app.js (hunk 9 above)
Comment (CosmicCoder96) [inline app/static/js/events/wizard/vue-app.js]: If one ticket is valid, why would the function exit?
if the ticket is valid, the `if` condition will not be satisfied, and the loop will keep moving forward.
Even if a single ticket is flawed then the function will exit and the button will be disabled.

**Comment by CosmicCoder96** [comment]
@niranjan94 I have made rest of the changes, can you please clarify about the logic being flawed once.

File: app/static/js/events/wizard/vue-app.js [hunk 11]
```diff
@@ -350,10 +357,22 @@ function shouldDisableMove($this) {
...
+        checkTickets($this.event.tickets)
     );
 }
 
+function checkTickets($this) {
+    var flag = false;
+    $this.forEach(function(ticket){
+        if(ticket.type === 'paid' && (ticket.price === '' || ticket.price <=0))
+        {
+            flag = true;
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: Okay ... this is fine ... I had accidentally read this as `return true` while reviewing .. Sorry about that :)

File: app/templates/gentelella/users/events/wizard/basic-details.html [hunk 12]
```diff
@@ -263,10 +263,11 @@ <h3 style="font-size: 15px;">{{ _("Quantity") }} <span class="required">*</span>
                         <h3 style="font-size: 15px;">{{ _("Options") }}</h3>
                     </div>
                 </div>
-                <div class="tickets-holder" style="margin-bottom: 10px;">
```
Comment (niranjan94) [inline app/templates/gentelella/users/events/wizard/basic-details.html]: Why has this class been removed ?

**Comment by CosmicCoder96** [comment]
@niranjan94 it's fixed now :) the class was removed accidentally.

File: app/static/js/events/wizard/vue-app.js [hunk 13]
```diff
@@ -171,6 +177,7 @@ var app = new Vue({
             ticket.sales_start_time = moment().tz(this.event.timezone).format('HH:mm');
             ticket.sales_end_date = moment().tz(this.event.timezone).add(10, 'days').format('MM/DD/YYYY'),
             ticket.sales_end_time = moment().tz(this.event.timezone).add(10, 'days').hour(22).minute(0).format('HH:mm'),
+            ticket.price = 0.01,
```
Comment (niranjan94) [inline app/static/js/events/wizard/vue-app.js]: There's a comma here instead of a semi-colon 

@CosmicCoder96

Comment (CosmicCoder96) [inline app/static/js/events/wizard/vue-app.js]: @niranjan94 there are commas on the above two lines as well , they were their originally too, Should I change them too?

**Comment by CosmicCoder96** [comment]
@niranjan94 done. Removed other commas as well.

**Pull request merged**